backend/services/theme_service.py

```
@@ -77,97 +77,343 @@ def _create_smart_person_background_mask(self, image_bytes: bytes, image_width:
+            mask_bytes = mask_buffer.getvalue()
+            
+            # Validate output
+            if len(mask_bytes) == 0:
+                raise RuntimeError("Generated mask is empty")
+            
+            logger.info(f"🧠 SMART PERSON-BACKGROUND MASK: {image_width}x{image_height} | Person colors analyzed | Background separated | Smooth blending applied | Mask size: {len(mask_bytes)/1024:.1f}KB")
+            return mask_bytes
+            
+        except Exception as e:
+            logger.error(f"❌ Mask generation failed during final processing: {str(e)}")
+            raise RuntimeError(f"Failed to generate smart person-background mask: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)